src/doc/rustc/src/platform-support.md

```diff
@@ -192,6 +195,8 @@ target | std | notes
 `x86_64-unknown-redox` | ✓ | Redox OS
 [`x86_64-unknown-uefi`](platform-support/unknown-uefi.md) | * | 64-bit UEFI
 
+[^x86_32-floats-x87]: Floating-point support on `i586` targets is non-compliant: the `x87` registers and instructions used for these targets do not provide IEEE-754-compliant behavior, in particular when it comes to rounding and NaN payload bits. See [issue #114479][x86-32-float-issue].
```

Comment:
One thing that's currently not mentioned is that even on these tier 1 targets, with the right flags you can run into the x87 issue -- e.g. by setting `-C target-cpu=pentium` on a i686 target. I'm not sure what the status is of such flags (is that even still a tier 1 target?) and how to best put this into the docs.